### PR TITLE
Add Comments For Getting Peer Cert Full Chain

### PR DESCRIPTION
**Pull request by ZhenLian**
This aims to solve part of the problem described in https://github.com/grpc/grpc/issues/25878.

### CODE REVIEW

File: include/grpc/grpc_security_constants.h [hunk 1]
```diff
@@ -29,6 +29,15 @@ extern "C" {
 #define GRPC_X509_CN_PROPERTY_NAME "x509_common_name"
 #define GRPC_X509_SAN_PROPERTY_NAME "x509_subject_alternative_name"
 #define GRPC_X509_PEM_CERT_PROPERTY_NAME "x509_pem_cert"
+// Please note that internally, we just faithfully pass whatever value we got by
+// calling SSL_get_peer_certificate() in OpenSSL/BoringSSL. This will mean in
```
Comment (jiangtaoli2016) [inline include/grpc/grpc_security_constants.h:33]: Actually, the code calls `SSL_get_peer_cert_chain()`.

File: include/grpc/grpc_security_constants.h [hunk 2]
```diff
@@ -29,6 +29,15 @@ extern "C" {
...
+// OpenSSL, the following conditions might apply:
+// 1. on the server side, the full chain would be the whole chain without the
+// leaf certificate
```
Comment (jiangtaoli2016) [inline include/grpc/grpc_security_constants.h:36]: I would rephrase

One the client side, this property returns the full certificate chain. On the server side, this property will return the certificate chain without the leaf certificate. Application can use GRPC_X509_PEM_CERT_PROPERTY_NAME to get the peer leaf certificate.

File: include/grpc/grpc_security_constants.h [hunk 3]
```diff
@@ -29,6 +29,15 @@ extern "C" {
...
+// 2. when session resumption happens, the full chain would be empty
+// BoringSSL wouldn't have this issue. For more, please refer to the official
+// OpenSSL manual:
+// https://www.openssl.org/docs/man1.0.2/man3/SSL_get_peer_certificate.html.
```
Comment (jiangtaoli2016) [inline include/grpc/grpc_security_constants.h:40]: nit: use this link https://www.openssl.org/docs/man1.1.0/man3/SSL_get_peer_cert_chain.html

File: include/grpc/grpc_security_constants.h [hunk 4]
```diff
@@ -29,6 +29,15 @@ extern "C" {
 #define GRPC_X509_CN_PROPERTY_NAME "x509_common_name"
 #define GRPC_X509_SAN_PROPERTY_NAME "x509_subject_alternative_name"
 #define GRPC_X509_PEM_CERT_PROPERTY_NAME "x509_pem_cert"
+// Please note that internally, we just faithfully pass whatever value we got by
+// calling SSL_get_peer_certificate() in OpenSSL/BoringSSL. This will mean in
+// OpenSSL, the following conditions might apply:
+// 1. on the server side, the full chain would be the whole chain without the
+// leaf certificate
+// 2. when session resumption happens, the full chain would be empty
```
Comment (jiangtaoli2016) [inline include/grpc/grpc_security_constants.h:37]: On this bullet, we can say

If the session is resumed, this property could be empty for OpenSSL (but not for BoringSSL).

File: src/core/tsi/ssl_transport_security.h [hunk 5]
```diff
@@ -40,6 +40,15 @@ extern "C" {
 
 #define TSI_X509_PEM_CERT_PROPERTY "x509_pem_cert"
 
+// Please note that internally, we just faithfully pass whatever value we got by
```
[the remark this answers is not in the document]
Comment (ZhenLian) [reply src/core/tsi/ssl_transport_security.h]: OK, I removed it.
I just thought adding a comment here would be a "reminder" for all other languages, otherwise if the field is exposed by other languages, we will need to remember to add similar comments.

File: include/grpc/grpc_security_constants.h (hunk 1 above)
Comment (ZhenLian) [reply include/grpc/grpc_security_constants.h]: Sorry, I was referring to the wrong link. Thanks for pointing it out!

**Review by jiangtaoli2016 — APPROVED** [review]
